### PR TITLE
add "mark as reported" to results table

### PR DESCRIPTION
**Pull request by FLX-0x00**
Reference: #1226 
The small icon in the result view (reported column) is now clickable if the email is not marked as reported. On click you have to confirm a dialogue which will mark the selected email as reported directly. We are facing some situations where this feature can be useful for example if a user report a phishing attempt via phone call. Tested on our servers, works great!

### CODE REVIEW

File: static/js/src/app/campaign_results.js [hunk 1]
```diff
@@ -925,7 +925,26 @@ function refresh() {
...
+    if (r == true) {
+        api.campaignId.get(cid).success((function(c) {
+            report_url = c.url;
+            report_url += report_url.endsWith("/") ? "" : "/";
+            report_url += "report?rid=" + rid;
+            $.ajax({
+                url: report_url,
+                method: "GET",
+                success: function(data) {
+                    return; // wait for response
```
Comment (jordan-wright) [inline static/js/src/app/campaign_results.js]: What do you think about moving the call to refresh here, that way it fires as soon as the `success` callback is fired?

**Comment by jordan-wright** [comment]
Hey there!

I'll give this a shot over the next couple of days. Thanks for sending it in!

Could you do me a favor and remove the .min.js file? I prefer to generate those myself since they're difficult to review in a PR.

I've also added a little comment that might make it such that the `refresh` call will fire a bit faster vs using `setTimeout`. Let me know what you think!

File: static/js/src/app/campaign_results.js (hunk 1 above)
Comment (blacklist-arcc) [reply static/js/src/app/campaign_results.js]: ofc, that will be a better code style! The timout / sleep function is important for the refresh to work, not the best solution i think, but works good. I will make a commit

**Review by FLX-0x00 — COMMENTED** [review]
ofc, that will be a better code style! The timout / sleep function is important for the refresh to work, not the best solution i think, but works good. Without the timout, the refresh is not working...I will make a commit.

**Comment by jordan-wright** [comment]
Doing some local testing, I didn't see that the `setTimeout` call is needed. And it _shouldn't_ be needed, since the `success` callback only fires after the call to `/report` is completed. So I'd recommend removing it.

And one last thing - would we please change the confirmation to leverage the SweetAlert2 library? This is how we do confirmations in other parts of Gophish. You can find an example of that here https://github.com/gophish/gophish/blob/master/static/js/src/app/campaign_results.js#L128

Let me know if you have any questions! This will be a good stop gap until we can figure out a longer term solution for doing a number of actions per recipient (resending emails, etc.)

**Comment by blacklist-arcc** [comment]
All right. Sound nice... I'll give it a shot and make some tests in the next days/pentests. Thanks for your input!

**Comment by blacklist-arcc** [comment]
Done. Tested in productive environment. Works as expected without any issues.

**Comment by jordan-wright** [comment]
LGTM - thank you!

**Comment by blacklist-arcc** [comment]
The strict Content-Security-Policy do not allow ajax requests from https to http. If you are using gophish with ssl and your landingpage not using ssl, the report-request will fail. So be sure to setup all your resources with https.

**Pull request merged**